### PR TITLE
Allow updating of Kubernetes cluster during reconciliation

### PR DESCRIPTION
**Pull request by mkjpryor**

### CODE REVIEW

**Review by sd109 — DISMISSED** [review]
LGTM

File: ui/src/components/pages/tenancy/platforms/kubernetes/card.js [hunk 1]
```diff
@@ -602,15 +601,15 @@ const KubernetesClusterDetailsButton = ({
...
+                                            disabled={inFlight || kubernetesCluster.status === "Deleting"}
                                             className="me-2"
                                         />
                                         <UpgradeKubernetesClusterButton
                                             kubernetesCluster={kubernetesCluster}
                                             kubernetesClusterActions={kubernetesClusterActions}
                                             kubernetesClusterTemplates={kubernetesClusterTemplates}
                                             kubernetesClusterTemplateActions={kubernetesClusterTemplateActions}
-                                            disabled={inFlight || working}
+                                            disabled={inFlight || kubernetesCluster.status.endsWith("ing")}
```
Comment (JohnGarbutt) [inline ui/src/components/pages/tenancy/platforms/kubernetes/card.js:612]: The problem they hit was being unable to move to a non-broken template. It had to be done manually.

Comment (mkjpryor) [reply ui/src/components/pages/tenancy/platforms/kubernetes/card.js:612]: I like less the idea of allowing an upgrade of a cluster that is still reconciling...

Comment (mkjpryor) [reply ui/src/components/pages/tenancy/platforms/kubernetes/card.js:612]: Specifically, the case I am worried about is:

  * User deploys cluster at version N
  * While it is still reconciling, user upgrades to N+1
  * While that is still reconciling, user upgrades to N+2
 
At that point, we potentially are violating the version skew policy.

**Review by JohnGarbutt — APPROVED** [review]
I like those upgrade checks, very cool.

**Pull request merged**